WithdrawalTransaction.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * WithdrawalTransaction class handles withdrawal operations.
+ */
+public class WithdrawalTransaction extends BaseTransaction {
+
+    public WithdrawalTransaction(double amount, Calendar date, String transactionID) {
+        super(amount, date, transactionID);
+    }
+
+    /**
+     * Applies the withdrawal to the given bank account.
+     * 
+     * @param ba the bank account from which the withdrawal is applied
+     * 
+     * Pre-condition:
+     * - The BankAccount object must not be null.
+     * - The bank account balance must be greater than or equal to the withdrawal amount.
+     * 
+     * Post-condition:
+     * - The bank account's balance decreases by the withdrawal amount, or an exception is thrown.
+     */
+    @Override
```

Comment:
Good job with this.

So in the specification you can add another annotation @throws InsufficientFundsException

Reply:
@Vankalondu,
can you try resolve the issues?